dataflow/src/main/java/org/checkerframework/dataflow/analysis/BackwardAnalysisImpl.java

```diff
@@ -0,0 +1,386 @@
+                        final SpecialBlockType sType = sBlock.getSpecialType();
+                        // storage the store at entry
+                        if (sType == SpecialBlockType.ENTRY) {
+                            storeAtEntry = outStores.get(sBlock);
+                        } else {
+                            assert sType == SpecialBlockType.EXIT
+                                    || sType == SpecialBlockType.EXCEPTIONAL_EXIT;
+                            for (Block pred : sBlock.getPredecessors()) {
+                                propagateStoresTo(
+                                        pred, null, getInput(sBlock), FlowRule.EACH_TO_EACH, false);
+                            }
+                        }
+                        break;
+                    }
+
+                default:
+                    throw new BugInCF(
+                            "BackwardAnalysisImpl::performAnalysis() unexpected block type: "
+                                    + block.getType());
+            }
+        }
+
+        if (!isRunning) {
+            throw new BugInCF(
+                    "BackwardAnalysisImpl::performAnalysis() when just finished the analysis loop"
+                            + " on Worklist, isRunning flag is expected to be true!");
+        }
+        isRunning = false;
+    }
+
+    @Override
+    public TransferInput<V, S> getInput(Block b) {
+        return inputs.get(b);
+    }
+
+    @Override
+    public S getEntryStore() {
+        return storeAtEntry;
+    }
+
+    @Override
+    protected void initFields(ControlFlowGraph cfg) {
+        super.initFields(cfg);
+        outStores = new IdentityHashMap<>();
+        exceptionStores = new IdentityHashMap<>();
+        // storeAtEntry is null before analysis begin
+        storeAtEntry = null;
+    }
+
+    @Override
+    protected void initInitialInputs() {
+        SpecialBlock regularExitBlock = cfg.getRegularExitBlock();
+        SpecialBlock exceptionExitBlock = cfg.getExceptionalExitBlock();
+
+        if (worklist.depthFirstOrder.get(regularExitBlock) == null
+                && worklist.depthFirstOrder.get(exceptionExitBlock) == null) {
+            throw new BugInCF(
+                    "regularExitBlock and exceptionExitBlock should never both be null at the same time.");
+        }
+
+        UnderlyingAST underlyingAST = cfg.getUnderlyingAST();
+        List<ReturnNode> returnNodes = cfg.getReturnNodes();
+
+        S normalInitialStore = transferFunction.initialNormalExitStore(underlyingAST, returnNodes);
+        S exceptionalInitialStore = transferFunction.initialExceptionalExitStore(underlyingAST);
+
+        // exceptionExitBlock and regularExitBlock will always be non-null, as
+        // CFGBuilder#CFGTranslationPhaseTwo#process()
+        // will always create these two exit blocks on a CFG no matter this CFG whether would has
+        // these exit blocks according to the underlying AST
```

Comment:
will always create these two exit blocks on a CFG whether it has exit blocks or not according to the underlying AST.